inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -571,22 +571,34 @@ def normalize_journal_titles(obj, eng):
     Returns:
        None
     """
-    publications = obj.data.get('publication_info')
-
-    if not publications:
-        return None
-
-    for index, publication in enumerate(publications):
-        if 'journal_title' in publication:
-            normalized_title = normalize_journal_title(publication['journal_title'])
-            obj.data['publication_info'][index]['journal_title'] = normalized_title
-            ref_query = RecordMetadata.query.filter(
-                RecordMetadata.json['_collections'].op('?')('Journals')).filter(
-                cast(RecordMetadata.json['short_title'], String) == type_coerce(normalized_title, JSON))
-            result = db.session.execute(ref_query).fetchone()
-
-            if result:
-                obj.data['publication_info'][index]['journal_record'] = result.records_metadata_json['self']
+    publications = obj.data.get('publication_info', [])
+
+    for publication in publications:
+        normalize_journal_title_entry(publication)
+
+    references = obj.data.get("references", [])
+    for reference in references:
+        publication_info = get_value(reference, 'reference.publication_info')
+        if not publication_info:
+            continue
+        normalize_journal_title_entry(publication_info)
```

Comment:
Not sure it's worth it, but it would be better to build a map `journal_title`->`(normalized_title, normalized_record)` to avoid having to lookup the same journal title multiple times. Even better would be to allow normalizing all titles with one query only, which is much more efficient.

Reply:
Idea is nice but I'm not sure how much we will benefit form it here, as we want in nearest future move workflows from NEXT so this will be rewritten as it will probably won't have access to DB at all.

Reply:
Even on ES it's inefficient to do multiple queries where one would suffice (`normalize_journal_title` is doing an ES query under the hood).

Reply:
Maybe we could create a new task to do this on ES in one query and not on db like now? This (maybe) will be useful for new implementation of workflows also.

Reply:
ok, can you create a task then?

Reply:
https://github.com/inspirehep/inspirehep/issues/1723